sdk/log/macro/src/lib.rs

```diff
@@ -235,3 +235,59 @@ pub fn log(input: TokenStream) -> TokenStream {
         TokenStream::from(quote! {pinocchio_log::logger::log_message(#format_string.as_bytes());})
     }
 }
+
+/// Attribute macro for instrumenting functions with compute unit logging.
+///
+/// This macro wraps the decorated function with additional logging statements
+/// that print the function name and the number of compute units used before and after
+/// the function execution.
+///
+/// # Effects
+///
+/// - Adds a log message with the function name at the end of execution with amount of CU consumed.
+///
+/// # Note
+///
+/// This macro consumes an additional compute units per call due to the logging operations.
+///
+///  # Example
+///
+/// ```rust,ignore
+/// #[pinocchio_log::compute_fn]
+/// fn my_function() {
+///     // Function body
+/// }
+/// ```
+/// 
+/// logging output will look like:
+/// 
+/// "Program log: Function my_function consumed 36 compute units"
+/// 
+/// # References
+///
+/// * [Logging syscall](https://github.com/anza-xyz/agave/blob/d88050cda335f87e872eddbdf8506bc063f039d3/programs/bpf_loader/src/syscalls/logging.rs#L70)
+/// * [Compute budget](https://github.com/anza-xyz/agave/blob/d88050cda335f87e872eddbdf8506bc063f039d3/program-runtime/src/compute_budget.rs#L150)
+///
+#[proc_macro_attribute]
+pub fn compute_fn(_attr: TokenStream, item: TokenStream) -> TokenStream {
+    let mut input = parse_macro_input!(item as ItemFn);
+    let fn_name = &input.sig.ident;
+    let block = &input.block;
+
+    input.block = syn::parse_quote!({
+        let cu_before =  ::pinocchio::log::sol_remaining_compute_units();
```

Comment:
`pinocchio_log` can be used in any program and currently does not depend on pinocchio. We need to add the syscall definition at the top the `logger.rs` file instead. This duplication will eventually go away when we integrate the SDK, but for now it is needed.

I can't comment directly there, but you will need to add:
```rust
pub fn sol_remaining_compute_units() -> u64;

pub fn sol_remaining_compute_units() -> u64 {
    let syscall: extern "C" fn() -> u64 = unsafe { core::mem::transmute(3991886574u64) };
    syscall(dest, src, n)
}
```